modules/caddyhttp/reverseproxy/healthchecks.go

```diff
@@ -153,32 +153,26 @@ func (h *Handler) doActiveHealthCheckForAllHosts() {
 					log.Printf("[PANIC] active health check: %v\n%s", err, debug.Stack())
 				}
 			}()
-			networkAddr := upstream.Dial
-			addr, err := caddy.ParseNetworkAddress(networkAddr)
-			if err != nil {
-				h.HealthChecks.Active.logger.Error("bad network address",
-					zap.String("address", networkAddr),
-					zap.Error(err),
-				)
-				return
-			}
-			if addr.PortRangeSize() != 1 {
-				h.HealthChecks.Active.logger.Error("multiple addresses (upstream must map to only one address)",
-					zap.String("address", networkAddr),
-				)
-				return
-			}
-			hostAddr := addr.JoinHostPort(0)
-			if addr.IsUnixNetwork() {
+
+			portStr := strconv.Itoa(upstream.activeHealthCheckPort)
+			hostAddr := net.JoinHostPort(upstream.networkAddress.Host, portStr)
+			if upstream.networkAddress.IsUnixNetwork() {
 				// this will be used as the Host portion of a http.Request URL, and
 				// paths to socket files would produce an error when creating URL,
 				// so use a fake Host value instead; unix sockets are usually local
 				hostAddr = "localhost"
 			}
-			err = h.doActiveHealthCheck(DialInfo{Network: addr.Network, Address: hostAddr}, hostAddr, upstream.Host)
+
+			dialInfo := DialInfo{
+				Network: upstream.networkAddress.Network,
+				Host:    upstream.networkAddress.Host,
+				Port:    portStr,
+				Address: hostAddr,
+			}
```

Comment:
How does this handle unix sockets? Does port end up becoming 0, and is that a problem? Can you add a simple test with unix sockets (or try it out)?

Reply:
This config works successfully, if this is what you mean:
```caddyfile
{
	http_port     9080
	https_port    9443
}
http://localhost:2020 {
	respond "Hello, World!"
}
http://localhost:2021 {
	bind unix/caddy-listener.sock
	respond "ok"
}
http://localhost:9080 {
	reverse_proxy {
		to localhost:2020

		health_path /health
		health_port 2021
		health_interval 2s
		health_timeout 5s
	}
}
```
If this is correct, I can add the test for it.
But

Reply:
Sorry; to clarify, I mean a proxy upstream that is addressed by unix socket rather than a port.

Reply:
Your question uncovered a bug :) Caddy v2.1.1 with unix socket upstream says:
```
2020/09/16 23:12:28.707 INFO    http.handlers.reverse_proxy.health_checker.active       HTTP request failed     {"host": "localhost", "error": "Get \"http://localhost:2021/health\": dial unix localhost: connect: no such file or directory"}
```

It dials `localhost` instead of the unix socket. I have a fix, I think :) Cleaning up the code, adding a test, and pushing a commit in a few.

Reply:
It should be working fine now